### PR TITLE
Merge ecc in development

### PR DESCRIPTION
**Pull request by andrei-diaconescu**
- Added ECC integration tests + gas costs fixes

### CODE REVIEW

File: integrationTests/vm/arwen/testdata/c-api-tests/cbuiltins.c [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    char *csrc = (char *)src;
+    char *cdest = (char *)dest;
+
+    for (unsigned long i = 0; i < n; i++)
+    {
+        cdest[i] = csrc[i];
+    }
+
+    return dest;
+}
```
Comment (bogdan-rosianu) [inline integrationTests/vm/arwen/testdata/c-api-tests/cbuiltins.c:22]: empty line at the end of the file?

File: integrationTests/vm/arwen/testdata/c-api-tests/crypto/elrond.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+    "language": "clang",
+    "source_files": [
+        "cryptoTest.c",
+        "../cbuiltins.c"
+    ]
+}
```
Comment (bogdan-rosianu) [inline integrationTests/vm/arwen/testdata/c-api-tests/crypto/elrond.json:7]: empty line at the end of the file?

File: integrationTests/vm/arwen/testdata/c-api-tests/ecBenchmark/ecBenchmark.export [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+p384UnmarshalCompressedEcTest
+p521UnmarshalCompressedEcTest
+p224GenerateKeyEcTest
+p256GenerateKeyEcTest
+p384GenerateKeyEcTest
+p521GenerateKeyEcTest
+p224ScalarMultEcTest
+p256ScalarMultEcTest
+p384ScalarMultEcTest
+p521ScalarMultEcTest
```
Comment (bogdan-rosianu) [inline integrationTests/vm/arwen/testdata/c-api-tests/ecBenchmark/ecBenchmark.export:47]: empty line at the end of the file?

File: integrationTests/vm/arwen/testdata/c-api-tests/ecBenchmark/elrond.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+    "language": "clang",
+    "source_files": [
+        "ecBenchmark.c",
+        "../cbuiltins.c"
+    ]
+}
```
Comment (bogdan-rosianu) [inline integrationTests/vm/arwen/testdata/c-api-tests/ecBenchmark/elrond.json:7]: empty line at the end of the file?

**Review by bogdan-rosianu — DISMISSED** [review]

File: integrationTests/vm/arwen/testdata/c-api-tests/ecBenchmark/ecBenchmark.c [hunk 5]
```diff
@@ -0,0 +1,856 @@
...
+typedef byte P256PRIVKEY[32];
+typedef byte P384PRIVKEY[48];
+typedef byte P521PRIVKEY[66];
+
+typedef byte P224MRESULT[57];
+typedef byte P256MRESULT[65];
+typedef byte P384MRESULT[97];
+typedef byte P521MRESULT[133];
+
+const int p224SMResultLength = 28;
```
Comment (AdoAdoAdo) [inline integrationTests/vm/arwen/testdata/c-api-tests/ecBenchmark/ecBenchmark.c:14]: please remove these ...Length constants if not used

**Review by AdoAdoAdo — CHANGES_REQUESTED** [review]

File: integrationTests/vm/arwen/testdata/c-api-tests/ecBenchmark/ecBenchmark.c [hunk 6]
```diff
@@ -0,0 +1,856 @@
...
+        0xcb, 0x40, 0xd6, 0x4a, 0x50, 0x1f, 0xba, 0x9c, 
+        0x11, 0x84, 0x7b, 0x28, 0x96, 0x5b, 0xc7, 0x37, 
+        0x7f, 0xff, 0xff, 0xc0, 0x3f, 0xff, 0xc0, 0x03, 
+        0xff, 0xff, 0xfc, 0x00, 0x7f, 0xff, 0x00, 0x00, 
+        0x7f, 0xff, 0xff, 0xc0, 0x3f, 0xff, 0xc0, 0x03, 
+        0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff
+    };
+
+const int numberOfReps = 10000;
+const int lengthOfScalar = 60;
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply integrationTests/vm/arwen/testdata/c-api-tests/ecBenchmark/ecBenchmark.c:45]: I see you use this for different curves, so I suspect you need different scalar lengths
why not give a parameter to repeatDoubleAndScalarMult for the scalar lenght, and pass that depending on the curve.

**Review by AdoAdoAdo — DISMISSED** [review]

**Review by sasurobert — DISMISSED** [review]

**Review by bogdan-rosianu — DISMISSED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**